modules/@apostrophecms/ui/ui/apos/components/AposButtonSplit.vue

```diff
@@ -133,11 +139,22 @@ export default {
       }
       this.setButton(initial);
     },
-    focus() {
-      // takes a moment to be on screen and focusable
-      setTimeout(() => {
-        this.$refs.choices[0].focus();
-      }, 200);
+    trapFocus() {
+      const selectedElementIndex = this.menu.findIndex(i => i.action === this.action) || 0;
+
+      // use map to keep items order:
+      this.elementsToFocus = this.menu.map(
+        i => this.$refs.choices.find(choice => choice.value === i.action)
+      );
```

Comment:
with `this.elementsToFocus = this.$refs.choices`, elements were not following the order in the DOM, leading to elements unfocusable and losing the focus.